scripts/controlnet.py

```diff
@@ -790,6 +790,10 @@ def postprocess(self, p, processed, *args):
         gc.collect()
         devices.torch_gc()
 
+
+# Do this so the preprocessor list is available for the API
+global_state.cn_preprocessors_names = Script().preprocessor
```

Comment:
As there are gradio components created in `__init__`, I am not sure we should be creating a script here. I think we should move the list of preprocessors and unloadables directly to `global_state` instead of replacing the dict here with an updated one.

Reply:
Note that doing this will create a bug for people that want to upgrade without restarting the webui completely. This is because the [`importlib.reload`](https://github.com/Mikubill/sd-webui-controlnet/pull/623/files#diff-86d1966226c5d294b3ed140211dcc872a2e27312f5f9b88c454e4717ef96f596L21-L22) calls are out of order. They should be:

```
importlib.reload(global_state)
importlib.reload(external_code)
```

Instead of the other way around as they are currently.

Reply:
No problem.

I did not see the migration section. Thank you. I will check the /controlnet/txt2img. 
I had not updated automatic 1111, everything works fine. Thank you.

Reply:
I will update the code. Hang on.